pkg/controllers/repositoryserver/repositoryserver_controller_test.go

```diff
@@ -134,24 +180,282 @@ func (s *RepoServerControllerSuite) CreateKopiaTLSSecret(data map[string][]byte)
+		{
+			description: "Invalid Repository Password",
+			testFunction: func(rs *v1alpha1.RepositoryServer) {
+				InvalidRepositoryPassword, err := s.CreateRepositoryPasswordSecret(testutil.GetRepoPasswordSecretData("invalidPassword"))
+				c.Assert(err, IsNil)
+
+				rs.Spec.Repository.PasswordSecretRef.Name = InvalidRepositoryPassword.Name
+				rs.Spec.Repository.PasswordSecretRef.Namespace = InvalidRepositoryPassword.Namespace
+			},
+		},
+		{
+			description: "Invalid Storage Location",
+			testFunction: func(rs *v1alpha1.RepositoryServer) {
+				storageLocationData := testutil.GetDefaultS3CompliantStorageLocation()
+				storageLocationData[repositoryserver.BucketKey] = []byte("invalidbucket")
+
+				InvalidStorageLocationSecret, err := s.CreateStorageLocationSecret(storageLocationData)
+				c.Assert(err, IsNil)
+
+				rs.Spec.Storage.SecretRef.Name = InvalidStorageLocationSecret.Name
+				rs.Spec.Storage.SecretRef.Namespace = InvalidStorageLocationSecret.Namespace
+			},
+		},
+		{
+			description: "Invalid Storage location credentials",
+			testFunction: func(rs *v1alpha1.RepositoryServer) {
+				storageLocationCredsData := testutil.GetDefaultS3StorageCreds()
+				storageLocationCredsData[secrets.AWSAccessKeyID] = []byte("testaccesskey")
+
+				InvalidStorageLocationCrdesSecret, err := s.CreateStorageLocationSecret(storageLocationCredsData)
+				c.Assert(err, IsNil)
+
+				rs.Spec.Storage.CredentialSecretRef.Name = InvalidStorageLocationCrdesSecret.Name
+				rs.Spec.Storage.CredentialSecretRef.Namespace = InvalidStorageLocationCrdesSecret.Namespace
+			},
+		},
+	} {
+		invalidCR := *originalrepoServerCR
+		tc.testFunction(&invalidCR)
+
+		repoServerCRCreated, err := s.crCli.RepositoryServers(s.repoServerControllerNamespace).Create(ctx, &invalidCR, metav1.CreateOptions{})
+		c.Assert(err, IsNil)
+
+		err = s.waitOnRepositoryServerState(c, repoServerCRCreated.Name)
+		c.Assert(err, NotNil)
+
+		//Get repository server CR with the updated server information
+		repoServerCRCreated, err = s.crCli.RepositoryServers(s.repoServerControllerNamespace).Get(ctx, repoServerCRCreated.Name, metav1.GetOptions{})
+		c.Assert(err, IsNil)
+
+		c.Assert(repoServerCRCreated.Status.Progress, Equals, v1alpha1.Failed)
+	}
+}
+
+func (s *RepoServerControllerSuite) waitForRepoServerInfoUpdateInCR(repoServerName string) error {
+	ctxTimeout := 25 * time.Minute
+	ctx, cancel := context.WithTimeout(context.Background(), ctxTimeout)
+	defer cancel()
+	err := poll.Wait(ctx, func(ctx context.Context) (bool, error) {
+		repoServerCR, err := s.crCli.RepositoryServers(s.repoServerControllerNamespace).Get(ctx, repoServerName, metav1.GetOptions{})
+		if err != nil {
+			return false, err
+		}
+		if repoServerCR.Status.ServerInfo.PodName == "" || repoServerCR.Status.ServerInfo.ServiceName == "" {
+			return false, nil
+		}
+		return true, nil
+	})
+	if err != nil {
+		return errors.Wrapf(err, "podname and service name are not set on repository server CR")
+	}
+	return err
```

Comment:
```suggestion
		if repoServerCR.Status.ServerInfo.PodName == "" || repoServerCR.Status.ServerInfo.ServiceName == "" {
			return false, errors.New("pod name or service name is not set on repository server CR")
		}
		return true, nil
	})
	if err != nil {
		return errors.Wrapf(err, "failed waiting for RepoServer Info updates in the CR")
	}
	return err
```

Reply:
sorry, this could have been highlighted in earlier reviews itself.

Reply:
The poll function stops executing when it gets an error. We dont want to return an error when the polling is on and pod name and server name are not set, since it takes some time to set these fields while the service and pod are being created, if we return error at first check, the test is always going to fail. I have
implemented it in some other way to get the exact error message, you can check that